api/api_calls.py

```diff
@@ -13,6 +13,10 @@
 from urllib.error import URLError
 
 import model
+# For a truly independent api module, we should have a signal, or pubsub system in the module, that the progress module
+# can subscribe to. That way, the api module is seperate, and other applications could use the emits/messages in their
+# own setups.
+import progress
```

Comment:
Making a truly separate signaling system is probably not needed at this point. If/when we want to pull the api module out and into it's own pip package then having it separated will need to be done.